dev/devenv-readme.md

```diff
@@ -12,49 +12,175 @@ First off, if you're on one of the following platforms, we provide the [devenv-i
-* [Eclipse Mars](https://eclipse.org/mars/)
-    * These Eclipse Plugins:
-        * [m2e-apt](https://developer.jboss.org/en/tools/blog/2012/05/20/annotation-processing-support-in-m2e-or-m2e-apt-100-is-out?_sscc=t): Allows Maven projects in Eclipse to easily leverage [the JDK's Annotation Processing framework](http://docs.oracle.com/javase/7/docs/technotes/guides/apt/).
 * [Apache Maven](https://maven.apache.org/)
+* [Latest Eclipse Version](https://www.eclipse.org/downloads/)
+* Eclipse __m2e-apt__ Plugin
+    * The m2e-apt plugin allows Maven projects in Eclipse to easily leverage [the JDK's Annotation Processing framework](http://docs.oracle.com/javase/7/docs/technotes/guides/apt/).
+    * Install m2e-apt using the [Drag To Install](http://marketplace.eclipse.org/content/m2e-apt) option or install from within the Eclipse IDE by opening **Help > Eclipse Marketplace...** and searching for **m2e-apt** using the **Eclipse Marketplace** find dialog box.  Click the **Install** button for the plugin and restart Eclipse when prompted. 
+    * [Usage Instructions for m2e-apt](https://developer.jboss.org/en/tools/blog/2012/05/20/annotation-processing-support-in-m2e-or-m2e-apt-100-is-out?_sscc=t)
 
 If you're not using one of those supported platforms, or would prefer to setup things manually, you'll want to download and install the items listed above yourself.
 
+## Cygwin Configuration
+
+### Associating .sh files with Cygwin
+Shell scripts that end in .sh should be associated with the Cygwin shell you have chosen to use.  Follow these steps to change the association for your installation:
+
+  1. Open a Command Prompt by pressing the Windows Key+R, type "cmd" and press return.
+  1. In the newly opened command prompt type:
+
+    ``` 
+    # Everything after .sh= is the ftype name
+    > assoc .sh
+    .sh=sh_auto_file
+    ``` 
+  1. Using the ftype name discovered in the previous step check the current association:
+
+    ```
+    # Example of sh_auto_file set to use git-bash
+    > ftype sh_auto_file
+    sh_auto_file="C:\Git\git-bash.exe" --no-cd "%L" %*
+    ```
+  1. If necessary, change the .sh association to use your Cygwin shell:
+
+    ```
+    # Example setting Cygwin bash to be associated with .sh files 
+    > ftype sh_auto_file="C:\cygwin64\bin\bash.exe" %1 %*
+    ```
+
+## AWS Configuration
+
+### AWS Credentials
+
+Many of the automated tests associated with the Blue Button framework use AWS resources.  Before running a build using Maven or importing projects into your Eclipse IDE, which will run a build automatically, please ensure the appropriate accounts and credentials are configured within your environment.  **This is necessary to prevent incurring unwanted charges on the wrong AWS account**.
+
+Below are links to detailed instructions on configuring your AWS credentials for your environment:
+
+  * [Configuring the AWS CLI](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html)
```

Comment:
I'd mention that this is optional -- nothing we do requires it. The credentials need to be setup, but the AWS Java libs use them in our builds; our builds don't use the AWS CLI.

Reply:
Its just a higher link, but I will remove it to eliminate any confusion.